From: Lennart Poettering <lennart@poettering.net>
Date: Sun, 1 Nov 2015 21:49:19 +0100
Subject: sd-daemon: verify NOTIFY_SOCKET path length

Better generate a real error then simply connect to the wrong socket.
---
 src/libsystemd/sd-daemon/sd-daemon.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/libsystemd/sd-daemon/sd-daemon.c b/src/libsystemd/sd-daemon/sd-daemon.c
index d2f10cd..9be321c 100644
--- a/src/libsystemd/sd-daemon/sd-daemon.c
+++ b/src/libsystemd/sd-daemon/sd-daemon.c
@@ -432,6 +432,11 @@ _public_ int sd_pid_notify_with_fds(pid_t pid, int unset_environment, const char
                 goto finish;
         }
 
+        if (strlen(e) > sizeof(sockaddr.un.sun_path)) {
+                r = -EINVAL;
+                goto finish;
+        }
+
         fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0);
         if (fd < 0) {
                 r = -errno;
